processor/metricstransformprocessor/metrics_transform_processor_test_helper.go

```diff
@@ -0,0 +1,738 @@
+			transforms: []Transform{
+				{
+					MetricName: metric1,
+					Action:     Insert,
+					NewName:    newMetric1,
+				},
+			},
+			in:  []testMetric{initialMetricRename1},
+			out: []testMetric{initialMetricRename1, outMetricRename1},
+		},
+		{
+			name: "metric_name_insert_multiple",
+			transforms: []Transform{
+				{
+					MetricName: metric1,
+					Action:     Insert,
+					NewName:    newMetric1,
+				},
+				{
+					MetricName: metric2,
+					Action:     Insert,
+					NewName:    newMetric2,
+				},
+			},
+			in:  []testMetric{initialMetricRename1, initialMetricRename2},
+			out: []testMetric{initialMetricRename1, initialMetricRename2, outMetricRename1, outMetricRename2},
+		},
+		{
+			name: "metric_label_update_with_metric_insert",
+			transforms: []Transform{
+				{
+					MetricName: metric1,
+					Action:     Insert,
+					NewName:    newMetric1,
+					Operations: []Operation{validUpateLabelOperation},
+				},
+			},
+			in:  []testMetric{initialMetricLabelRename1},
+			out: []testMetric{initialMetricLabelRename1, outMetricLabelRenameInsert1},
+		},
+		{
+			name: "metric_label_value_update_with_metric_insert",
+			transforms: []Transform{
+				{
+					MetricName: metric1,
+					Action:     Insert,
+					NewName:    newMetric1,
+					Operations: []Operation{validUpdateLabelValueOperation},
+				},
+			},
+			in:  []testMetric{initialLabelValueRename1},
+			out: []testMetric{initialLabelValueRename1, outLabelValueRenameInsert1},
+		},
+		{
+			name: "metric_label_aggregation_sum_int_insert",
+			transforms: []Transform{
+				{
+					MetricName: metric1,
+					Action:     Insert,
+					Operations: []Operation{validUpdateLabelAggrSumOperation},
+				},
+			},
+			in:  []testMetric{initialAggrBuilder(metricspb.MetricDescriptor_GAUGE_INT64)},
+			out: []testMetric{initialAggrBuilder(metricspb.MetricDescriptor_GAUGE_INT64), outLabelAggrBuilder(4, metricspb.MetricDescriptor_GAUGE_INT64)},
+		},
+		{
+			name: "metric_label_values_aggregation_sum_int_insert",
+			transforms: []Transform{
+				{
+					MetricName: metric1,
+					Action:     Insert,
+					Operations: []Operation{validUpdateLabelValuesAggrSumOperation},
+				},
+			},
+			in:  []testMetric{initialAggrBuilder(metricspb.MetricDescriptor_GAUGE_INT64)},
+			out: []testMetric{initialAggrBuilder(metricspb.MetricDescriptor_GAUGE_INT64), outLabelValuesAggrBuilder(4, metricspb.MetricDescriptor_GAUGE_INT64)},
+		},
+		{
+			name: "metric_label_values_aggregation_sum_distribution_insert",
+			transforms: []Transform{
+				{
+					MetricName: metric1,
+					Action:     Insert,
+					Operations: []Operation{validUpdateLabelAggrSumOperation},
+				},
+			},
+			in:  []testMetric{initialDistValueMetricBuilder()},
+			out: []testMetric{initialDistValueMetricBuilder(), outDistValueMetricBuilder()},
+		},
+	}
+)
+
+// constructTestInputMetricsDate builds the actual metrics from the test cases
+func constructTestInputMetricsData(test metricsTransformTest) consumerdata.MetricsData {
```

Comment:
Also, what would you think about making the transforms be actual YAML configuration? That way the test cases are a form of education about how to use these aggregations and it verifies that your parsing logic is correct.

Reply:
@james-bebbington I will go some research on this and try it out. @draffensperger I think I want to have this test more focused on the functionality of the processing because I have already tested the parsing logic in the config_test.go. Let me know if I am missing anything.

Reply:
Yea I was thinking yaml would be nice a well, but not totally sure it's worth the extra effort / code. If you start with the builder pattern, this could always be extended to read from yaml later